cpp/src/io/utilities/datasource.cpp

```diff
@@ -227,14 +228,27 @@ class memory_mapped_source : public file_source {
   }
 
  private:
+  [[nodiscard]] bool should_register_mmap_buffer()
+  {
+    if (_map_addr == nullptr) { return false; }
+
+    auto const policy = getenv_or("LIBCUDF_MMAP_REGISTER_ENABLED", std::string{"AUTO"});
+
+    if (policy == "ALWAYS") { return true; }
+    if (policy == "AUTO") { return pageableMemoryAccessUsesHostPageTables(); }
+    if (policy == "OFF") { return false; }
```

Comment:
Should we ignore cases?

Reply:
We could. All other env vars are case sensitive, so to me it makes sense to have a separate PR where we switch to case-insensitive across the board. I can improve the env var utils in that PR.